### PR TITLE
Sync confirm fligths with DB

### LINKED ISSUE #285: synch confirm flights with DB
on confirming a flight:
- transform planned flight into confirmed flight
- synch the confirmed flight with the DB

### PR DESCRIPTION
**Pull request by Prolas**
Confirming a flight updates the entry in the database. The old planned flight is replaced with a confirmed flight.

close #285

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/skysync/components/ConfirmFlightUITest.kt [hunk 1]
```diff
@@ -201,25 +202,42 @@ class ConfirmFlightUITest {
...
   fun confirmButtonWorksWhenDismiss() = runTest {
     composeTestRule.onNodeWithTag("AlertDialog").assertIsNotDisplayed()
+
+    composeTestRule.onNodeWithTag("LazyList").performScrollToNode(hasText("Enter Remark"))
+    composeTestRule.onNodeWithText("Confirm").assertDoesNotExist()
+
+    val setTime = composeTestRule.onAllNodesWithText("Set Time")
+    for (i in 0 until setTime.fetchSemanticsNodes().size) {
+      setTime[i].performClick()
+    }
```
Comment (Zep3k) [inline app/src/androidTest/java/ch/epfl/skysync/components/ConfirmFlightUITest.kt:212]: Shouldn't you scroll to those elements first before performing an action on them?

**Review by Zep3k — APPROVED** [review]
LGTM

File: app/src/androidTest/java/ch/epfl/skysync/components/ConfirmFlightUITest.kt [hunk 2]
```diff
@@ -201,25 +202,42 @@ class ConfirmFlightUITest {
...
     composeTestRule.onNodeWithTag("AlertDialogDismiss").performClick()
     composeTestRule.onNodeWithTag("AlertDialog").assertIsNotDisplayed()
 
-    // Test that navigate was not called (the route is not meaningful)
-    verify(exactly = 0) { navController.navigate(Route.MAIN) }
+    verify(exactly = 0) { navController.navigate(Route.HOME) }
   }
 
   @Test
   fun confirmButtonWorksWhenConfirm() = runTest {
```
Comment (mathieumognol) [inline app/src/androidTest/java/ch/epfl/skysync/components/ConfirmFlightUITest.kt:224]: A bit confusing with the precedent function as they lookalike except for one word

File: app/src/androidTest/java/ch/epfl/skysync/screens/ConfirmFlightScreenTest.kt [hunk 3]
```diff
@@ -49,23 +50,46 @@ class ConfirmFlightScreenTest {
...
-      composeTestRule
-          .onNodeWithTag("LazyList")
-          .performScrollToNode(hasText("Confirm"))
-          .assertIsDisplayed()
+
+      composeTestRule.waitUntil(2500) {
+        composeTestRule.onAllNodesWithText("Balloon").fetchSemanticsNodes().isNotEmpty()
+      }
+
+      composeTestRule.onNodeWithTag("LazyList").performScrollToNode(hasText("Enter Remark"))
```
Comment (mathieumognol) [inline app/src/androidTest/java/ch/epfl/skysync/screens/ConfirmFlightScreenTest.kt:83]: 3rd copy of that code block between this test and the previous one, could've made a helper function to call when needed

File: app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt [hunk 4]
```diff
@@ -53,13 +54,27 @@ fun confirmation(plannedFlight: PlannedFlight, confirmClick: () -> Unit) {
   val date: LocalDate = plannedFlight.date
   val timeSlot = plannedFlight.timeSlot
   val vehicles: List<Vehicle> = plannedFlight.vehicles
+  val meetupLocationPassenger = "Nancy"
```
Comment (mathieumognol) [inline app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt:57]: Weird to still have hardcoded values in this component but maybe it was already here

**Review by mathieumognol — APPROVED** [review]
LGTM

File: app/src/androidTest/java/ch/epfl/skysync/components/ConfirmFlightUITest.kt (hunk 1 above)
Comment (Prolas) [reply app/src/androidTest/java/ch/epfl/skysync/components/ConfirmFlightUITest.kt:212]: It is not the best way to do it but `composeTestRule.onNodeWithTag("LazyList").performScrollToNode(hasText("Enter Remark")` ensures that all Set Times are displayed

File: app/src/androidTest/java/ch/epfl/skysync/components/ConfirmFlightUITest.kt (hunk 2 above)
Comment (Prolas) [reply app/src/androidTest/java/ch/epfl/skysync/components/ConfirmFlightUITest.kt:224]: I agree that the difference is small. I will merge them in 1 function where I check first the dismiss button and then confirm

File: app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt (hunk 4 above)
Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt:57]: Yes it was already there. I agree we will have to change it asap

File: app/src/androidTest/java/ch/epfl/skysync/screens/ConfirmFlightScreenTest.kt (hunk 3 above)
Comment (Prolas) [reply app/src/androidTest/java/ch/epfl/skysync/screens/ConfirmFlightScreenTest.kt:83]: I don't think a function for one line is necessary.

**Pull request merged**